controllers/index.js

```diff
@@ -2,35 +2,267 @@
+			model.continueMessage = res.bundle.get('keepShopping');
+
+		}
+		else {
+			//Ready the products for display
+			Object.keys(sessionCart).forEach(function (itemid) {
+				var item = sessionCart[itemid];
+				cartLength += item.qty;
+				displayCart.items.push(item);
+				total += (item.qty * item.price);
+				totalItems += item.qty;
+			});
 
+			req.session.total = displayCart.total = total.toFixed(2);
+			model.items = displayCart;
+			model.totalItems = totalItems;
+			model.itemsInCart = res.bundle.getIntl('items', {cartItemLength: cartLength});
+		}
 
-	router.get('/', function (req, res) {
+		callback(null, model);
+	};
 
+	var getProducts = function (callback) {
 		Product.find(function (err, prods) {
 			if (err) {
 				console.log(err);
 			}
-			prods.forEach(function (prod) {
-				prod.prettyPrice = prod.prettyPrice();
+			prods = prods.map(function (prod) {
+				prod.prettyPrice = prod.prettifyPrice();
+				return prod;
 			});
+			var staticProd = [{
+				prettyPrice: '$100.00',
+				name: 'foo',
+				id: 1
+			}];
 			var model = {
 				products: prods,
 				title: 'Kraken Store'
 			};
-			res.render('index', model);
+			//console.log('model', model);
+			callback(null, model);
 		});
+	};
+	async.parallel({cart: getCart, products: getProducts}, function (err, result) {
+		var model = {};
 
+		model.products = result.products.products;
+		model.cart = result.cart;
+		//console.log('model', model);
+		callback(null, model);
 	});
+}
+module.exports.cart = function (req, res, next) {
 
-	router.get('/setLanguage/:locale', function (req, res) {
+	//Load (or initialize) the cart
+	req.session.cart = req.session.cart || {};
+	var cart = req.session.cart;
+
+	//Read the incoming product data
+	var id = req.param('item_id');
+
+	//Locate the product to be added
+	Product.findById(id, function (err, prod) {
+		if (err) {
+			console.log('Error adding product to cart: ', err);
+			return next(err);
+		}
+
+		//Add or increase the product quantity in the shopping cart.
+		if (cart[id]) {
+			cart[id].qty++;
+		}
+		else {
+			cart[id] = {
+				name: prod.name,
+				price: prod.price,
+				prettyPrice: prod.prettifyPrice(),
+				qty: 1
+			};
+		}
+		modelBuilder(req, res, function (err, model) {
+			if (err) {
+				return next(err);
+			}
+			res.status(200).json(model);
+		});
+
+
+	});
+};
+module.exports.setLocale = function (req, res) {
 		var localeString = req.params.locale;
 		var language = localeString.split('-')[1];
 		var country = localeString.split('-')[0];
 		var localeObject = {language: language, country: country};
 		res.cookie('locale', localeObject);
 		res.redirect('/');
+	};
+
+module.exports.newProduct = function (req, res) {
+	var name = req.body.name && req.body.name.trim();
+
+	//***** PLEASE READ THIS COMMENT ******\\\
+	/*
+	 Using floating point numbers to represent currency is a *BAD* idea \\
+
+	 You should be using arbitrary precision libraries like:
+	 https://github.com/justmoon/node-bignum instead.
```

Comment:
this comment is copied over from the original code. i'm more focused on the react changes right now but that's a good point.